tests/storages_tests/test_journal.py

```diff
@@ -116,3 +118,21 @@ def test_pop_waiting_trial_multiprocess_safe() -> None:
                 if trial_id is not None:
                     trial_id_set.add(trial_id)
         assert len(trial_id_set) == num_enqueued
+
+
+def test_pickle_dump_and_load() -> None:
+    with tempfile.TemporaryDirectory() as workdir:
+        file_name = os.path.join(workdir, "journal.log")
+        file_storage = optuna.storages.JournalFileStorage(file_name)
```

Comment:
How about parametrizing the log storage using `LOG_STORAGE`? Note that the current `JournalRedisStorage` is not pickalable, so we need to skip the test for it.